home/models.py

```diff
@@ -32,6 +33,12 @@ class Profile(models.Model):
     def __str__(self):
         return '{self.user.username}'.format(self=self)
 
+    @classmethod
+    def create(cls, username, password, email, gender='U', is_blocked=False):
```

Comment:
The DB operations below probably aught to be done in a transaction together

Why arn't you saving the profile object?